win32ss/gdi/eng/gradient.c

```diff
@@ -251,7 +254,7 @@ IntEngGradientFillRect(
   FINITCOL(linefrom, lineto, 0); \
   FINITCOL(linefrom, lineto, 1); \
   FINITCOL(linefrom, lineto, 2); \
-  for(g = sx[linefrom]; g != sx[lineto]; g += gxi) \
+  for(g = sx[linefrom] - 1; g != sx[lineto] + gxi; g += gxi) \
```

Comment:
This change fill most of the gaps that were present after commit 1.

Reply:
I understand the `+ gxi`, but can you explain the first `- 1`?
For performance reasons the calculation should be pulled outside of the loop.
```
    g_start = sx[linefrom] - 1; \
    g_end = sx[lineto] + gxi; \
    for (g = g_start; g != g_end; g += gxi) \
```

Reply:
Regarding the ```- 1```. This fixes the white diagonal line from top-right to bottom-left in many of the examples by starting the line one pixel sooner.
Now I have implemented your suggestion of using ```g_start``` and ```g_end```. Thanks.